### PR TITLE
Users deletion (backend)

### PR DESCRIPTION
**Pull request by EvanAntoni**
User delete feature. Pull request into branch users_list.

### CODE REVIEW

File: easyrest/models/user.py [hunk 1]
```diff
@@ -70,3 +70,12 @@ def add(database, form_data):
...
         database.add(User(name=name, email=email, password=password_hash))
+
+    @staticmethod
+    def delete(database, user):
+        """This function removes the user and his data from the database.
+
+        :param database: Database session.
+        :param user: Instance of deletable user.
+        """
+        database.delete(user)
```
Comment (MaxSukhovetskyi) [inline easyrest/models/user.py]: In this particular case you can do delete method like this:

def delete(self, session):
      session.delete(self)

So no need to write @staticmethod

**Review by MaxSukhovetskyi — APPROVED** [review]

File: easyrest/models/user.py (hunk 1 above)
Comment (EvanAntoni) [reply easyrest/models/user.py]: It makes sense. Thank's.
But what's wrong with the`@staticmethod`?

Comment (MaxSukhovetskyi) [reply easyrest/models/user.py]: Its confusing)

Comment (EvanAntoni) [reply easyrest/models/user.py]: Guess which function deletes the user?
Function 1:
```
def delete(self, session):
    session.delete(self)
```
Function 2:
```
def delete(database, user):
    database.delete(user)
```

**Review by RazSKg — APPROVED** [review]

File: easyrest/models/user.py (hunk 1 above)
Comment (serge30) [reply easyrest/models/user.py]: Both of them are confusing. I do not see any reasons for such functions, as they do not use any context. You may just inline its body in place of use.

Using in controller something like is acceptable and clear:
```
request.dbsession.delete(user)
```

Comment (EvanAntoni) [reply easyrest/models/user.py]: I made this function with some expectation for the future. For example, if you need to delete files associated with a user (well, avatar or something else), you can simply do something like this to the delete method:
```
def delete(database, user):
    storage.delete(user.avatar)
    database.delete(user)
```

**Pull request merged**